### PR TITLE
[bb] check if symlink instead of whether resolved filename is "bb"

### PR DESCRIPTION
**Pull request by lsiudut**
The logic of resolving symlinks until we encounter "bb" introduced in
aaa8821 also broke a use cases when the
binary generated with makebb is not named "bb" after all. This can be
easily acheved by calling makebb with -o switch.

```
$ go build
$ ./makebb -o test ../../cmds/{ls,df}
$ ln -s makebb ls
lrwxrwxrwx 1 lsiudut lsiudut       4 Jan 30 11:18 ls -> test
-rwxr-xr-x 1 lsiudut lsiudut 8929697 Jan 30 11:14 makebb
-rw-r--r-- 1 lsiudut lsiudut    1112 Jan 30 11:12 makebb.go
-rwxr-xr-x 1 lsiudut lsiudut 2290720 Jan 30 11:18 test
$ ./ls
2019/01/30 11:18:10 os.Args is [test]: you need to specify which command to invoke.
```

I propose to check if resolved path is a symlink and continue traversal if it is. If
not - assume that this is our final bb binary.

### CODE REVIEW

**Comment by rminnich** [comment]
This seems reasonable but I'd like to see what the others think.

Although, IMHO, I don't see a reason to enable names other than bb

**Comment by lsiudut** [comment]
Sure, but in this case it would be good to document it and remove `-o` switch from makebb.

We got caught in this trap because we were using `makebb` in different project than u-root.

File: pkg/bb/cmd/main.go [hunk 1]
```diff
@@ -19,9 +19,17 @@ func run() {
 	}
 }
 
+func isSymlink(f string) bool {
+	s, err := os.Stat(f)
+	if err != nil {
+		return false
+	}
+	return s.Mode()&os.ModeSymlink != 0
```
Comment (rminnich) [inline pkg/bb/cmd/main.go]: actually, I would be a bit more careful here. While this is right now, there are cases where more than one bit is set to indicate a file type and it's burned people.
return (s.Mode()&os.ModeType) == os.ModeSymlink

this is similar to IsRegular()
I realize your form works however, I just feel safer with the full compare.

**Comment by hugelgupf** [comment]
Hey - I'd like to take this, I'm in the same situation.

@lsiudut can you fix the DCO (developer certificate of origin) and decide whether to incorporate ron's comment or not?

**Comment by lsiudut** [comment]
Yes sir, sorry for the radio silence.

I actually remember experimenting with IsRegular, don't remember exactly why I decided to go with just-link explicit comparison. Let me recap...

**Comment by lsiudut** [comment]
Ok, there's quite some commotion with signing off the diffs. Checks were complaining that my first diff here was also not signed so I rebased back to it and signed it too, but not it requires me to sign all the diffs (not mine) that landed due to rebase.

Do you want me to simply open new clean PR? I guess it would be the cleanest solution.

**Comment by lsiudut** [comment]
Ok, here's my lame restart of this PR: https://github.com/u-root/u-root/pull/1140. Because sign requirement was introduced in the meantime and it got rebased not-by-me a first diff got stuck far behind diffs from rebase and I couldn't find a way of signing it without making system think that I also need to sign bunch of non-mine diffs.

**Pull request closed without merging**